File: modules/api/src/main/java/com/natevaughan/kchat/api/ChatRepo.java
package com.natevaughan.kchat.api;

import java.util.Collection;

/**
 * Created by nate on 12/9/17
 */
public interface ChatRepo extends Repository<Chat> {
	Chat findByKey(String key);
	Collection<Chat> findNewForUser(User user, Long timestamp);
    Collection<Chat> listForUser(User user);
}

File: modules/chat-api/src/main/kotlin/com/natevaughan/kchat/api/ChatRepo.kt
package com.natevaughan.kchat.api

/**
 * Created by nate on 12/9/17
 */
interface ChatRepo : Repository<Chat> {
	fun findNewForUser(user: User, timestamp: Long?): Collection<Chat>
	fun listForUserAndSpace(spaceId: String, user: User): Collection<Chat>
	fun addParticipant(chatId: String, userId: String): Boolean
	fun getPrivileges(chatId: String, userId: String): ContextualPrivilege?
	fun delete(chatId: String): Boolean
}